caikit_nlp/modules/embedding_retrieval/embedding.py

```diff
@@ -0,0 +1,143 @@
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .embedding_retrieval_task import EmbeddingRetrievalTask
+from caikit_nlp.data_model.embedding_vectors import Vector1D
+
+logger = alog.use_channel("<EMBD_BLK>")
+error = error_handler.get(logger)
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingRetrievalTask,
+)
+class EmbeddingModule(ModuleBase):
+
+    _ARTIFACTS_PATH_KEY = "artifacts_path"
+    _ARTIFACTS_PATH_DEFAULT = "artifacts"
+
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
+        """Load model
+
+        Args:
+            model_path: str
+                Path to the config dir under the model_id (where the config.yml lives)
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        artifacts_path = config.get(cls._ARTIFACTS_PATH_KEY)
+
+        error.value_check(
+            "<NLP07391618E>",
+            artifacts_path,
+            ValueError(f"Model config missing '{cls._ARTIFACTS_PATH_KEY}'"),
+        )
+
+        artifacts_path = os.path.abspath(os.path.join(model_path, artifacts_path))
+        error.dir_check("<NLP34197772E>", artifacts_path)
+
+        return cls.bootstrap(model_name_or_path=artifacts_path)
+
+    def run(
+        self, input: str, **kwargs  # pylint: disable=redefined-builtin
+    ) -> Vector1D:
+        """Run inference on model.
+        Args:
+            input: str
+                Input text to be processed
+        Returns:
+            Vector1D: the output
+        """
+        error.type_check("<NLP27491611E>", str, input=input)
+
+        return Vector1D.from_embeddings(self.model.encode(input))
+
+    @classmethod
+    def bootstrap(cls, model_name_or_path: str) -> "EmbeddingModule":
+        """Bootstrap a sentence-transformers model
+
+        Args:
+            model_name_or_path: str
+                Model name (Hugging Face hub) or path to model to load.
+        """
+        return cls(model=SentenceTransformer(model_name_or_path=model_name_or_path))
+
+    def save(self, model_path: str, *args, **kwargs):
+        """Save model using config in model_path
+
+        Args:
+            model_path: str
+                Path to model config
+        """
+
+        model_config_path = model_path  # because the param name is misleading
+
+        error.type_check("<NLP82314992E>", str, model_path=model_config_path)
+        error.value_check(
+            "<NLP40145207E>",
+            model_config_path is not None and model_config_path.strip(),
+            f"model_path '{model_config_path}' is invalid",
+        )
+
+        model_config_path = os.path.abspath(
```

Comment:
The ModuleSaver docstring says it takes an "absolute path":
``` 
model_path (str): The absolute path to the directory where the model
           will be saved.  If this directory does not exist, it will be
           created.
```
It doesn't look like that is necessary today, but better to follow the doc.

Reply:
re: example using ModuleSaver...  I'd recommend you do not use the ModuleSaver context manager.  It isn't safe. https://github.com/caikit/caikit/issues/525